### PR TITLE
v1.0.0 alpha release

### PR DESCRIPTION
**Pull request by ryankurte**
🎉🎊🥳

See #177 for progress on blocking issues

### CODE REVIEW

**Comment by luojia65** [comment]
Should we deal with all design pattern issues and pull requests before a probable 1.0 release? After 1.0, they would be no longer changed.

**Comment by ryankurte** [comment]
Ideally yep, see #177 for the list, mostly predicated on coming to a consensus as to #172 (and #98) as you mentioned in the chat.

However, I don't think this necessarily needs to block an alpha release, and there's a balance to be struct somewhere between avoiding breaking releases while not blocking e-h improvements indefinitely :-/

**Comment by luojia65** [comment]
Ohh yes, do not need to block alpha here. Just block before the final 1.0.0 :)

**Comment by Disasm** [comment]
I think that the only thing that should block v1.0.0 is https://github.com/rust-embedded/wg/pull/435. We can add all the other changes after this release. `embedded-hal` project is already moving extremely slow, so it's better not to block it for one more reason. This async story began just recently and I do not expect it to end in the coming year.

Also I don't understand why we need `v1.0.0-alpha-1` and not just `v1.0.0`. We can use `v1.0.0` and `v1.0.1` instead if we expect any non-breaking changes soon.

**Comment by luojia65** [comment]
Yes, I agree. There should be no more blocks here

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,8 +5,18 @@ All notable changes to this project will be documented in this file.
...
 
+
 ## [Unreleased]
 
+### Added
+
+### Changed
+
+
+## [v1.0.0] - 2020-06-10
```
[the remark this answers is not in the document]
Comment (therealprof) [reply CHANGELOG.md:16]: That's fine, the big question then is how to do the CHANGELOG for the final 1.0.0 but we can also think about that when we get there. I always find it odd when the meat of a release is burried under some alpha/beta/pre-release and the final release only says: "and now it's final". ;)

[the remark this answers is not in the document]
Comment (ryankurte) [reply CHANGELOG.md:16]: oh yeah, i guess we could merge em all together in the final one

**Comment by ryankurte** [comment]
turns out i can't approve this because it's my pr, but, lgtm~!

**Pull request merged**